### ISSUE TITLE
WithFoo edge eager load query method doc doesn't say edges must not be null

### ISSUE THREAD

**Issue by willfaught** [ent/ent#4020]
For example:

```go
// WithChannelSubscriptions tells the query-builder to eager-load the nodes that are connected to
// the "channel_subscriptions" edge. The optional arguments are used to configure the query builder of the edge.
func (uq *UserQuery) WithChannelSubscriptions(opts ...func(*ChannelSubscriptionQuery)) *UserQuery {
```

I assumed if opts is nil, then rows with null ChannelSubscriptions would be included. The generated doc did not indicate otherwise.

Also, there is no doc I could find on the Ent website for what generated WithFoo methods do. That would help as well.

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

Generated code for eager loading a "ChannelSubscription" edge on type "User":

```go
// WithChannelSubscriptions tells the query-builder to eager-load the nodes that are connected to
// the "channel_subscriptions" edge. The optional arguments are used to configure the query builder of the edge.
func (uq *UserQuery) WithChannelSubscriptions(opts ...func(*ChannelSubscriptionQuery)) *UserQuery {
```

## Expected Behavior 🤔

Generated code for eager loading a "ChannelSubscription" edge on type "User":

```diff
 // WithChannelSubscriptions tells the query-builder to eager-load the nodes that are connected to
 // the "channel_subscriptions" edge. The optional arguments are used to configure the query builder of the edge.
+// Requires that the "channel_subscriptions" edge is not null.
 func (uq *UserQuery) WithChannelSubscriptions(opts ...func(*ChannelSubscriptionQuery)) *UserQuery {
```

## Steps to Reproduce 🕹

```go
// ent/schema/user.go

func (User) Edges() []ent.Edge {
	return []ent.Edge{
// ...
		edge.From("channel_subscriptions", ChannelSubscription.Type).
			Ref("users").
			Unique().
			Field("subscription_id"),
// ...
	}
}

// ent/schema/channelsubscription.go

func (ChannelSubscription) Edges() []ent.Edge {
	return []ent.Edge{
// ...
		edge.To("users", User.Type).
// ...
	}
}
```

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | go version go1.22.2 darwin/arm64  |
| Ent         | 0.13.1   |
| Database    | PostgresQL   |
| Driver      | github.com/lib/pq@v1.10.9        |

**Comment by a8m** [comment]
Hey! Please, see this doc: https://entgo.io/docs/eager-load

In short, `With<E>` queries all edges associated with the queried nodes (of the root builder). 

Closing, but please use our Discord server in case you have more questions regarding this API - it’s widely used by our users.

**Comment by willfaught** [comment]
Thanks for the link, but that page doesn't show that rows are excluded if the edges are null. In my opinion, it shows the opposite:

```go
    // Populate the `pets` that associated with the `admins`.
    WithPets().
```

It should say this:

```diff
-    // Populate the `pets` that associated with the `admins`.
+    // Populate the `pets` that associated with the `admins`,
+    // and exclude rows with null pets.
     WithPets().
```

**Comment by a8m** [comment]
Not sure where you get it from, but it doesn’t exclude nodes without edges. Eager-loaded edges are evaluated after the nodes are queried.

**Comment by willfaught** [comment]
I observed that behavior yesterday:

```go
return c.Ent.User.Query().
		WithChannelSubscriptions().
		Where(userFilters...).
		Paginate(ctx, after, first, before, last, paginationOptions...)
```

userFilters was nil. Pagination vars after, first, before, last, and paginationOptions were nil. 21 rows were missing from the result. 14 missing rows were accounted for by the soft delete mixin for the User type. The only hooks and interceptors on the User type are from the soft delete mixin, which was copied from the example on the website.

I concluded the other 7 missing rows were accounted for by their subscription_id field (corresponding to the ChannelSubscription type) being null. I didn't see any other possible explanation.
